frontend/README.md

```diff
@@ -49,65 +51,109 @@ WARNING: Do not store sensitive data in the environment variables.
 
 ## Deployment
 
-1.  Ensure your `awscli` is configured with the
-    [required credentials and profiles](../docs/awscli.md).
-    Set the appropriate `AWS_PROFILE`.
+1. Ensure your `awscli` is configured with the
+   [required credentials and profiles](../docs/awscli.md).
+   Set the appropriate `AWS_PROFILE`.
+
+   ```shell
+   export AWS_PROFILE=single-cell-dev
+   ```
+
+1. **Specify deployment.**
+
+   Set the `DEPLOYMENT_STAGE` environment variable to a valid deployed environment: `dev`, `staging`
+
+   ```shell
+   export DEPLOYMENT_STAGE=dev
+   ```
+
+1. **Deploy.**
+
+   Files are deployed to a publicly accessible bucket. Do not include sensitive data in the deployed files.
+
+   ```shell
+   make deploy
+   ```
+
+## Smoke Test
+
+### Update Tests
+
+#### Gatsby Dev Build
+
+1. Start FE server: `npm run develop`
+
+2. Run tests: `npm run e2e`
+
+#### Gatsby Prod Build
+
+1. Start FE server: `npm run build-and-start-prod`
+
+1. Run tests: `npm run e2e-localProd`
+
+### Local
+
+#### FE + Dev BE
+
+1. Requirements:
+
+   1. Your `src/configs/configs.js` points to Dev API URL instead of `localhost`. E.g., `https://api.dev.corpora.cziscience.com`
+
+1. For Gatsby Prod Build: `npm run smoke-test-prod-build`
 
-    ```shell
-    export AWS_PROFILE=single-cell-dev
-    ```
+1. For Gatsby Dev Build: `npm run smoke-test`
 
-1.  **Specify deployment.**
+#### FE + Local BE
 
-    Set the `DEPLOYMENT_STAGE` environment variable to a valid deployed environment: `dev`, `staging`
+1. Requirements:
 
-    ```shell
-    export DEPLOYMENT_STAGE=dev
-    ```
+   1. You are able to spin up a local BE server
+   1. Your `src/configs/configs.js` points to local BE server. E.g., `http://localhost:5000`
 
-1.  **Deploy.**
+1. For Gatsby Prod Build: `npm run smoke-test-with-local-backend`
 
-    Files are deployed to a publicly accessible bucket. Do not include sensitive data in the deployed files.
+### CI/CD
```

Comment:
Add more explanation. What does this list mean?

Reply:
Good point! Done!